lib/llm/LLMProvider.ts

```diff
@@ -77,3 +77,20 @@ export class LLMProvider {
     }
   }
 }
+
+export const resolveLLMClient = (
```

Comment:
This doesn't need to take in `llmClient`. Also, instead of ordered params, it would be better to have an object passed in with a function signature like `({ llmProvider, modelName, requestId }: { llmClient?: llmClient, modelName?: AvailableModel, requestId?: string })`